src/lxc/namespace.c

```diff
@@ -53,16 +53,17 @@ static int do_clone(void *arg)
 	return clone_arg->fn(clone_arg->arg);
 }
 
+#define __LXC_STACK_SIZE 4096
 pid_t lxc_clone(int (*fn)(void *), void *arg, int flags)
 {
+	size_t stack_size;
+	pid_t ret;
 	struct clone_arg clone_arg = {
-		.fn = fn,
-		.arg = arg,
+	    .fn = fn,
+	    .arg = arg,
 	};
-
-	size_t stack_size = lxc_getpagesize();
-	void *stack = alloca(stack_size);
-	pid_t ret;
+	char *stack[__LXC_STACK_SIZE] = {0};
```

Comment:
The commit message should probably explain that (and why) this also changes from `getpagesize()` to a hardcoded `4096`.

Reply:
Avoiding allocations